apiService.md

```diff
@@ -0,0 +1,77 @@
+
+### Key Differences
+
+- Axios needs to be imported
+
+```javascript
+    npm install axios
+```
+
+### apiServices
+
+Now, there is an `apiService.ts` file that takes care of all the headers in `src/apiService`. This file imports the `Resources` like Application, Attribute, Source, etc., from `/resources/`. It needs to import `axios , { AxiosInstance } from 'axios'`, along with the resources.
+
+It exports a class component with the tokens safely set privately and the `axiosClient` method to instantiate an HTTP request with `baseURL` set and Headers specified with tokens, content-type, and accepted returned formats.
+
+The apiServices call upon resources to instantiate
+
+### Resources
+
+The resources need instances (they get this from the apiService), this instance receives one of four methods (getAll, getOne, create, update). The resources are made separately for easy-to-read, maintainable, reusable components that allow custom calling. This way improves performance as if doesn't need to go through a maze of if-else statements.
+
+### apiContext
+
+To avoid having to call new instances everywhere in components that need instances, React solves this state issue with a `Provider` component. This Provider Component specifies data required elsewhere in the application without going heavy on Redux or Prop Drilling your state through unnecessary components.
+
+This component imports the `APIService`. Then [creates](https://reactjs.org/docs/context.html#reactcreatecontext) context with the APIService data we need to make globally accessible and store that in a variable.
+
+It exports this variable as a `Provider` for other components to import and use (as [`Consumers`](https://reactjs.org/docs/context.html#contextconsumer)).
+
+### Provider
+
+The Provider component needs to be imported in a top component, like `<App />` or often the `<Layout />` component (another wrapper component). In this case, call upon the `APIService` like:
+
+```javascript
+const API = new APIService(sessionStorage.getItem("jwt"));
+```
+
+> this example is in JavaScript
+
+The wrapping is enough to make this data globally accessible in your application when you add `API` as value props (`value={API}`) to the `APIProvider` component. The Provider in our application is the `apiContext` exported as `<APIProvider />` component.
```

Comment:
Misschien beter verwoord door te zeggen dat alles ín deze wrapper nu bij de API kan. Het is niet per se nu globally available (onderdelen buiten deze wrapper kunnen er namelijk niet bij).